### PR TITLE
Use fs-extra instead of mkdirp

### PR DESCRIPTION
**Pull request by AlCalzone**
fixes: #497 

Next step would be replacing all fs.xyzSync methods with the async counterparts from fs-extra.

### CODE REVIEW

File: lib/letsencryptStore.js [hunk 1]
```diff
@@ -1,98 +1,30 @@
...
-        return fs_unlinkAsync(bakname)
-            .then(() => fs_renameAsync(filename, bakname))
+        return fs.unlink(bakname)
+            .then(() => fs.rename(filename, bakname))
             // this must be successful
             .catch(_err =>
-                fs_renameAsync(tmpname, filename)
+                fs.rename(tmpname, filename)
                     .catch(err =>
                         sfs.revert(filename)
```
Comment (AlCalzone) [inline lib/letsencryptStore.js:29]: sfs.revert is not defined by the way.

**Comment by Apollon77** [comment]
Er leave that for 3.1. so DO NOT MERGE NOW!

**Comment by AlCalzone** [comment]
Sure. The issue was scheduled for 3.1, so I thought this was clear :)

**Comment by Apollon77** [comment]
@AlCalzone could you please rebase to fix the conflicts?

**Comment by AlCalzone** [comment]
Sure thing, but first I'll need to sleep 😅

**Comment by AlCalzone** [comment]
rebased

**Comment by Apollon77** [comment]
as said this is included in #799 too ... so I close this one

**Pull request merged**